Biern/Models/DataProvider.swift

```diff
@@ -14,28 +17,50 @@ class DataProvider: ObservableObject {
         var request = URLRequest(url: url)
         request.setValue("application/json", forHTTPHeaderField: "Content-Type")
         request.httpMethod = httpMethod
+        request.setValue("Bearer \(self.token)", forHTTPHeaderField: "Authorization")
         return request
     }
 
+    func login() {
```

Comment:
Might be good to reset reason and alert everytime a new request is fired because otherwise you might show your user errors which are triggered by a previous call